parts/windows/kuberneteswindowssetup.ps1

```diff
@@ -332,6 +332,7 @@ try
     # In case of AKSCustomCloud, customer cloud env will be loaded from azurestackcloud.json 
     {{if IsAKSCustomCloud}}
     $azureStackConfigFile = [io.path]::Combine($global:KubeDir, "azurestackcloud.json")
+    $env:AZURE_ENVIRONMENT_FILEPATH = $azureStackConfigFile

```

Comment:
Which components will use this environment variable in running AKS Windows CSE script in node provisioning?

Reply:
gotcha, I guess these are separate envs then

Reply:
we want to make sure this set after CSE finishes so the new secure TLS bootstrap client-go credential plugin will be able to use it in order to request a TLS bootstrap token to join the cluster. The main idea is that instead of using a hard-coded TLS bootstrap token, we'll instead point kubelet towards this new credential plugin (which is just a go binary) via bootstrap-kubeconfig in order to generate a dynamic and short-lived TLS bootstrap token instead

Reply:
I think that $env:AZURE_ENVIRONMENT_FILEPATH only can be used in current powershell session used by CSE. Please test whether it works as expected in the new secure TLS bootstrap client-go credential plugin

Reply:
Setting Environment Variable Persistently
To set the environment persistently so they should remain even when close the session, PowerShell uses [System.Environment] class with the SetEnvironmentVariable method for the environment variable to set it persistently.

[System.Environment]::SetEnvironmentVariable('ResourceGroup','AZ_Resource_Group')
PS C:\> $env:ResourceGroup
AZ_Resource_Group